### PR TITLE
docs(lsp): add enum type annotation `vim.lsp.protocol.Methods`

### PR DESCRIPTION
**Pull request by rcasia**
Added the enum type annotation `vim.lsp.protocol.Methods` to provide some intellisense support,
improving developer experience with autocomplete suggestions and documentation.

### CODE REVIEW

**Review by MariaSolOs — APPROVED** [review]

**Comment by mfussenegger** [comment]
You'd also need to run the script to update the `protocol.lua`, but I don't really see any change with lua-ls, with or without this change?

File: runtime/lua/vim/lsp/protocol.lua [hunk 1]
```diff
@@ -942,5 +943,14 @@ protocol.Methods = {
   --- The `workspace/workspaceFolders` is sent from the server to the client to fetch the open workspace folders.
   workspace_workspaceFolders = 'workspace/workspaceFolders',
 }
+local function freeze(t)
```
Comment (max397574) [inline runtime/lua/vim/lsp/protocol.lua:946]: why would you add this here?
perhaps rebased on wrong branch?

Comment (rcasia) [reply runtime/lua/vim/lsp/protocol.lua:946]: This was autogenerated when running `nvim -l scripts/gen_lsp.lua gen --methods`.

It is hardcoded in the script: https://github.com/neovim/neovim/blob/8f5e90811025bcc652146168d5116f8d73fe8f81/scripts/gen_lsp.lua#L114-L127

It seems someone forgot to run it in the past. For what I checked, every reference to lsp.Methods complies with this changes.

Comment (max397574) [reply runtime/lua/vim/lsp/protocol.lua:946]: 🤦‍♂️ I didn't realize that this also was part of the autogenerated things

**Comment by rcasia** [comment]
@mfussenegger I initially thought this would be handled by GitHub Actions, but it seems not. I've already updated it.

**Review by MariaSolOs — APPROVED** [review]

**Comment by rcasia** [comment]
@mfussenegger Hi, just following up on this PR. I’ve addressed all requested changes and am happy to assist further. Thanks for your time!

File: runtime/lua/vim/lsp/protocol.lua (hunk 1 above)
Comment (justinmk) [reply runtime/lua/vim/lsp/protocol.lua:946]: freeze was removed in https://github.com/neovim/neovim/commit/0a0962a2e8d9ffa5f03e492fa293e964650a965e , please regen and rebase.

Comment (rcasia) [reply runtime/lua/vim/lsp/protocol.lua]: Just completed the regen and rebase. A different set of changes was generated due to the regen script, but they seem appropriate.

**Pull request merged**